pkg/errors/sdk_errors.go

```diff
@@ -79,17 +77,19 @@ func CheckError(resp *http.Response, expectedStatusCodes ...int) SDKError {
 		}
 	}
 
-	var content map[string]interface{}
-	if err := json.NewDecoder(resp.Body).Decode(&content); err != nil {
-		return NewSDKErrorWithStatus(err, resp.StatusCode)
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return NewSDKErrorWithStatus(fmt.Errorf("%w : %s", errRespBody, err.Error()), resp.StatusCode)
```

Comment:
there is an internal package for handling errors which supports wrapping can we use that?
https://github.com/mainflux/mainflux/blob/master/pkg/errors/errors.go

Reply:
@mteodor , Changed to mainflux errors